### ISSUE TITLE
Is `long?` an integer type?

### ISSUE THREAD

**Issue by bzbarsky** [heycam/webidl#670]
Per spec, looks like no.  That means that `[Clamp] long?` should be an error, right?  Is that the intent?

@domenic

**Comment by peria** [comment]
Is there any rule to connection?  As far as I understand, it's not clear `[Clamp] long?` should be interpreted as `[Clamp] (long?)` or `([Clamp] long)?`.
IDL grammar defines it in `[Clamp] (long?)` format, but I think its interpretation does not need to follow the grammar.

**Comment by bzbarsky** [comment]
Right, that is exactly what this issue is about.  Per grammar, you end up with `[Clamp] (long?)` which as currently defined is an error.  It may make more sense to treat that case as valid and apply the clamping to the `long` in that situation, but it needs spec changes to accomplish that.

**Comment by bzbarsky** [comment]
Note that some people are trying to use things like this in their spec IDL, and that fails in Gecko, because we implement the spec as currently written.  It would be good to sort out exactly how this should work, so browsers don't end up creating incompatible spec extensions to handle this case...  @domenic, could you please take a look, since you probably have the best idea of what the original goal here was?
